database/src/main/dataset_schema/add_schema.sql

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+DROP FUNCTION IF EXISTS dataset_schema.add_schema(TEXT, INTEGER, TEXT, JSONB, TEXT);
+
+CREATE OR REPLACE FUNCTION dataset_schema.add_schema(
+    IN i_schema_name        TEXT,
+    IN i_schema_version     INTEGER,
+    IN i_schema_description TEXT,
+    IN i_fields             JSONB,
+    IN i_user_name          TEXT,
+    OUT status              INTEGER,
+    OUT status_text         TEXT,
+    OUT id_schema           BIGINT
+) RETURNS record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: jobs_configuration.add_schema(5)
+--      Stores a new version of the schema.
+--      The i_schema_version has to be an increment of the latest version of an existing schema or 1
+--
+-- Parameters:
+--      i_schema_name               - name of the schema
+--      i_schema_version            - version of the schema
+--      i_schema_description        - description of the schema
+--      i_fields                    - the fields the schema consist of
+--      i_user_name                 - the user who submitted the changes
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_schema           - id of the newly created schema record
+--
+-- Status codes:
+--      201     - OK
+--      403     - Schema already exists
+--      404     - Schema version wrong
+--      405     - Schema has been disabled
+--      406     - Schema is locked
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _latest_version INTEGER;
+    _locked         BOOLEAN;
+    _disabled       BOOLEAN;
+BEGIN
+    SELECT dsh.schema_latest_version, dsh.locked_when IS NOT NULL, dsh.disabled_when IS NOT NULL
+    FROM dataset_schema.heads dsh
+    WHERE dsh.schema_name = i_schema_name
```

Comment:
how is NULL here? Are we scared of the input being null or god forbid the name in DB?